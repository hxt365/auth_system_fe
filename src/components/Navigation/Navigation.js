import React, { useContext } from 'react';
import { Menu } from 'antd';
import { HOME, CHANGE_PASSWORD, LOGIN } from 'constants/route';
import { Link } from 'react-router-dom';
import { authServices } from 'services';
import { AppContext } from 'components/AppLayout';

function Navigation() {
  const state = useContext(AppContext);

  const logoutHandler = () => {
    authServices.logout();
    state.setUser({
      username: '',
      firstName: '',
      lastName: '',
      is_authenticated: false,
    });
  };

  return (
    <Menu
      theme="dark"
      mode="horizontal"
      defaultSelectedKeys={[HOME]}
      style={{ lineHeight: '6.4rem', float: 'right' }}
    >
      <Menu.Item key={HOME}>
        <Link to={HOME}>Home</Link>
      </Menu.Item>
      <Menu.Item key={CHANGE_PASSWORD}>
        <Link to={CHANGE_PASSWORD}>Change password</Link>
      </Menu.Item>
      <Menu.Item key={LOGIN}>
        <Link to={LOGIN} onClick={logoutHandler}>
          Logout
        </Link>
      </Menu.Item>
    </Menu>
  );
}

export default Navigation;
