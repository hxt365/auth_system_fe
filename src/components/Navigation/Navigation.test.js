import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

describe('test navigation', () => {
  it('test snapshot', () => {
    const wrapper = shallow(<Navigation />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
