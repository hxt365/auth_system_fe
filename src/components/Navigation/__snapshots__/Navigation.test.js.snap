// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test navigation test snapshot 1`] = `
<Menu
  defaultSelectedKeys={
    Array [
      "/",
    ]
  }
  mode="horizontal"
  style={
    Object {
      "float": "right",
      "lineHeight": "6.4rem",
    }
  }
  theme="dark"
>
  <MenuItem
    key="/"
  >
    <Link
      to="/"
    >
      Home
    </Link>
  </MenuItem>
  <MenuItem
    key="/password/change"
  >
    <Link
      to="/password/change"
    >
      Change password
    </Link>
  </MenuItem>
  <MenuItem
    key="/login"
  >
    <Link
      onClick={[Function]}
      to="/login"
    >
      Logout
    </Link>
  </MenuItem>
</Menu>
`;
